SmellDetector/Detectors/RuleEngines/GodClassRuleEngine.cs

```diff
@@ -0,0 +1,39 @@
+﻿using SmellDetector.Controllers;
+using SmellDetector.SmellModel;
+using SmellDetector.SmellModel.Reports;
+using System;
+
+namespace SmellDetector.Detectors
+{
+    internal class GodClassRuleEngine : IDetector
+    {
+        public PartialSmellDetectionReport FindIssues(CaDETClassDTO caDetClassDto)
+        {
+            PartialSmellDetectionReport partialReport = new PartialSmellDetectionReport();
+
+            foreach (var identifierAnalysis in caDetClassDto.CodeItemMetrics)
+            {
+
+                if (IsBadSmell(identifierAnalysis.Value))
+                {
+                    Issue newIssue = new Issue();
+                    newIssue.IssueType = SmellType.GOD_CLASS;
+                    newIssue.CodeItemId = identifierAnalysis.Key;
+                    partialReport.AddIssue(identifierAnalysis.Key, newIssue);
+                }
+
+            }
+
+            return partialReport;
+        }
+
+        private bool IsBadSmell(MetricsDTO metrics)
+        {
+            if (metrics.WMC > 47 && metrics.ATFD > 5 && metrics.TCC < 0.33) return true;
```

Comment:
We need a few tests to check if the detection strategies are correct.

Reply:
done :)